src/components/sol/swap.tsx

```diff
@@ -0,0 +1,558 @@
+  ]);
+
+  // effect to fetch quote when inputs change
+  React.useEffect(() => {
+    fetchSwapQuote();
+  }, [fetchSwapQuote]);
+
+  return (
+    <div className="mb-12 mt-4 flex w-full flex-col items-center justify-center gap-8">
+      <div className="rounded-lg border p-4">
+        <div className="flex flex-col items-center justify-center gap-4">
+          <TokenInput
+            assets={inAssets}
+            selectedAsset={tokenFrom}
+            amount={amountFrom}
+            disabled={isLoadingQuote}
+            onTokenSelect={(token) => {
+              reset();
+              setTokenFrom(token);
+            }}
+            onAmountChange={setAmountFrom}
+          />
+          <div className="flex gap-2">
+            <ArrowUpIcon size={18} />
+            <ArrowDownIcon size={18} />
+          </div>
+          <TokenInput
+            assets={outAssets}
+            selectedAsset={tokenTo}
+            amount={amountTo}
+            capMaxAmount={false}
+            showWalletBalance={false}
+            showQuickAmountButtons={false}
+            onTokenSelect={setTokenTo}
+            onSearch={onSearch}
+            disabled={true}
+          />
+        </div>
+        <div className="mt-4 flex justify-between">
+          <Button
+            variant="ghost"
+            size="sm"
+            className="gap-1.5 font-normal"
+            onClick={fetchSwapQuote}
+            disabled={
+              isLoadingQuote ||
+              !tokenFrom ||
+              !tokenTo ||
+              !amountFrom ||
+              amountFrom <= 0
+            }
+          >
+            <RefreshCwIcon
+              size={13}
+              className={isLoadingQuote ? "animate-spin" : ""}
+            />{" "}
+            Refresh
+          </Button>
+          <TxnSettings
+            trigger={
+              <Button variant="ghost" size="sm" className="gap-1.5 font-normal">
+                <SettingsIcon size={13} /> Settings
+              </Button>
+            }
+          />
+        </div>
+        {swapQuote && (
+          <div
+            className={cn(
+              "mt-3 space-y-2 border-t pt-4 text-xs",
+              isLoadingQuote && "animate-pulse",
+            )}
+          >
+            <div className="flex justify-between">
+              <span className="text-muted-foreground">Priority</span>
+              <span className="capitalize">{priority}</span>
+            </div>
+            <div className="flex justify-between">
+              <span className="text-muted-foreground">Slippage</span>
+              <span>{(swapQuote.slippageBps / 100).toFixed(2)}%</span>
+            </div>
+            <div className="flex justify-between">
+              <span className="text-muted-foreground">Price Impact</span>
+              <span
+                className={
+                  Number(swapQuote.priceImpactPct) > 1 ? "text-destructive" : ""
+                }
+              >
+                {priceImpact}%
```

Comment:
the priceImpact is denoted as for example as 0.02 which equals 2%

so I think you need to move it 2 decimals up if you want to display it correctly with %